src/main/java/software/bernie/techarium/network/ServerToClientPacket.java

```diff
@@ -0,0 +1,20 @@
+package software.bernie.techarium.network;
+
+import java.util.Optional;
+
+import net.minecraftforge.fml.network.NetworkDirection;
+import net.minecraftforge.fml.network.NetworkEvent.Context;
+
+public abstract class ServerToClientPacket<MSG extends ServerToClientPacket<MSG>> extends Packet<MSG>{
+
+	@Override
+	public boolean isValid(Context context) {
+		return getDirection().isPresent();
```

Comment:
you should return true, because the packet direction is already validated by forge and you could make getDirection final, so no serverpacket can override it.